File: Source/Aladdin.ASN1.ISO/ASN1/ISO/PKCS/PKCS05/PBES2Parameter.cs
using System;
using System.Runtime.Serialization;

//	PBES2Parameter ::= SEQUENCE {
//		keyDerivationFunc	AlgorithmIdentifier,
//		encryptionScheme	AlgorithmIdentifier
//	}

namespace Aladdin.ASN1.ISO.PKCS.PKCS5
{
	[Serializable]
	public class PBES2Parameter : Sequence
	{
		// информация о структуре
		private static readonly ObjectInfo[] info = new ObjectInfo[] { 

			new ObjectInfo(new ObjectCreator<AlgorithmIdentifier>().Factory(), Cast.N), 
			new ObjectInfo(new ObjectCreator<AlgorithmIdentifier>().Factory(), Cast.N), 
		}; 
		// конструктор при сериализации
        protected PBES2Parameter(SerializationInfo info, StreamingContext context) : base(info, context) {}

		// конструктор при раскодировании
		public PBES2Parameter(IEncodable encodable) : base(encodable, info) {}

		// конструктор при закодировании
		public PBES2Parameter(AlgorithmIdentifier keyDerivationFunc, 
			AlgorithmIdentifier encryptionScheme) : 
			base(info, keyDerivationFunc, encryptionScheme) {}

		public AlgorithmIdentifier	KeyDerivationFunc	{ get { return (AlgorithmIdentifier)this[0]; } }
		public AlgorithmIdentifier	EncryptionScheme	{ get { return (AlgorithmIdentifier)this[1]; } }
	}
}

File: Source/Aladdin.ASN1.ISO.Java/src/aladdin/asn1/iso/pkcs/pkcs5/PBES2Parameter.java
package aladdin.asn1.iso.pkcs.pkcs5;
import aladdin.asn1.*; 
import aladdin.asn1.iso.*; 
import java.io.*; 

//	PBES2Parameter ::= SEQUENCE {
//		keyDerivationFunc	AlgorithmIdentifier,
//		encryptionScheme	AlgorithmIdentifier
//	}

public final class PBES2Parameter extends Sequence<AlgorithmIdentifier>
{
    private static final long serialVersionUID = -8253854200958284542L;
    
	// информация о структуре
	private static final ObjectInfo[] info = new ObjectInfo[] { 

		new ObjectInfo(new ObjectCreator(AlgorithmIdentifier.class).factory(), Cast.N), 
		new ObjectInfo(new ObjectCreator(AlgorithmIdentifier.class).factory(), Cast.N), 
	}; 
	// конструктор при раскодировании
	public PBES2Parameter(IEncodable encodable) throws IOException { super(encodable, info); }

	// конструктор при закодировании
	public PBES2Parameter(AlgorithmIdentifier keyDerivationFunc, 
		AlgorithmIdentifier encryptionScheme) 
	{
		super(info, keyDerivationFunc, encryptionScheme); 
	}
	public final AlgorithmIdentifier	keyDerivationFunc() { return get(0); }
	public final AlgorithmIdentifier	encryptionScheme () { return get(1); }
}

